src/openzeppelin/upgrades/library.cairo

```diff
@@ -35,83 +35,91 @@ end
 # Initializer
 #
 
-func Proxy_initializer{
-        syscall_ptr: felt*,
-        pedersen_ptr: HashBuiltin*,
-        range_check_ptr
-    }(proxy_admin: felt):
-    let (initialized) = Proxy_initialized.read()
-    with_attr error_message("Proxy: contract already initialized"):
-        assert initialized = FALSE
+namespace Proxy:
+
+    #
+    # Initializer
+    #
+
+    func initializer{
```

Comment:
I think we should call it `constructor` like the rest of the libraries, no? Or at least open yet another discussion :p

Reply:
Haha so many discussions! Yeah, using `constructor` is probably best. I was thinking this was a special case, but it really operates in the same way the other constructor/initializers do

Reply:
In the latest [commits](https://github.com/OpenZeppelin/cairo-contracts/pull/300/commits/69681249040a98e07369619c16ef797cb4152e47), the Proxy library's `initializer` is changed to `constructor`, but I kept the `initializer` function name in the implementation contracts. My thought process is to ensure users don't think to use `initializer` in the decorated `@constructor`. See example [here](https://github.com/andrew-fleming/cairo-contracts/blob/namespace-upgrades/src/openzeppelin/token/erc20/ERC20_Upgradeable.cairo#L19-L36). Since it must be invoked after deployment, I think `initializer` is more appropriate

Reply:
Sounds good